### DISCUSSION TITLE
Nuxt 3 equivalent to Nuxt 2’s context.from?

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#18830] [Questions] [answered]
In Nuxt 2 we have this very [usefull context property .from](https://nuxtjs.org/docs/internals-glossary/context#from). Is there an equivalent for Nuxt 3?

If not how can I get it though?

**Comment by TheAlexLichter** [comment, marked as answer]
> In Nuxt 2 we have this very [usefull context property .from](https://nuxtjs.org/docs/internals-glossary/context#from). Is there an equivalent for Nuxt 3?
> 
> If not how can I get it though?

In which "context would you need it?

**Reply by ghost** [reply]
I am talking about this property https://nuxtjs.org/docs/internals-glossary/context/#from

I need it inside a back button component, because I want to know if the user comes from my app or from somewhere else.

**Reply by TheAlexLichter** [reply]
Yes, I saw the link. But I wanted to understand your use case 😋

There is no built-in parameter. You could go with the good old `router.back()` call on click, but that would not be an `<a>` tag then (but a button).

Also you could use the router's history state:

```
const router = useRouter()
console.log(router.options.history.state)
/* 
{
  "back": null,
  "current": "/",
  "forward": null,
  "position": 1,
  "replaced": true,
  "scroll": false
}
*/
```

Another option would be to keep track of the routes yourself in e.g. `app.vue` by watching them and saving them to an own stack (or just saving the last link). Problem is that you have to keep in mind the history pop/pushes

**Reply by ghost** [reply]
@manniL Thanks a lot. `router.options.history.state.back` is exactly what I need!
